File: Python/series.py
def slices(series, length):
    if series == [] or series == "":
        raise ValueError("series cannot be empty")
    else:
        if length < 0:
            raise ValueError("slice length cannot be negative")
        elif length == 0:
            raise ValueError("slice length cannot be zero")
        elif length > len(series):
            raise ValueError("slice length cannot be greater than series length")
        else:
            slices = [""]*(len(series) - length + 1)
            for index in range(len(slices)):
                slices[index] = series[index:(index+length)]
    return slices
    

File: Ruby/series.rb
class Series
    def initialize(serie)
      @series = serie
    end
  
    def slices(length)
      if length <= @series.length then
        subtrings = []
        for index in (0..@series.length-length) do
          subtrings.append(@series[index..index+length-1])
        end
      else
        raise ArgumentError
      end
    return subtrings
    end
  end
  